File: backend/src/main/java/com/eburtis/abproject/service/StatutService.java
package com.eburtis.abproject.service;

import com.eburtis.abproject.domain.Statut;
import com.eburtis.abproject.repository.StatutRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StatutService {

	private final StatutRepository statutRepository;

	public StatutService(StatutRepository statutRepository) {
		this.statutRepository = statutRepository;
	}

	/**
	 * Récupère tous les statuts disponibles
	 * 
	 * @return Liste de tous les statuts
	 */
	public List<Statut> getAllStatuts() {
		return statutRepository.findAll();
	}

	/**
	 * Récupère un statut par son code
	 * 
	 * @param codeStatut Code du statut
	 * @return Le statut trouvé
	 */
	public Statut getStatutByCode(String codeStatut) {
		return statutRepository.findByCodeStatut(codeStatut)
				.orElseThrow(() -> new RuntimeException("Statut non trouvé: " + codeStatut));
	}
}

File: frontend/src/services/statut.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Statut } from '../models/statut.model';
import { ChangementStatut } from '../models/changement-statut.model';
import { urls } from './api-urls';

@Injectable({
    providedIn: 'root'
})
export class StatutService {
    private apiUrl = urls.statuts;
    private changementsUrl = urls.changementsStatut;

    constructor(private http: HttpClient) {}

    /**
     * Récupère tous les statuts disponibles
     */
    getAllStatuts(): Observable<Statut[]> {
        return this.http.get<Statut[]>(this.apiUrl);
    }

    /**
     * Récupère un statut par son code
     */
    getStatutByCode(codeStatut: string): Observable<Statut> {
        return this.http.get<Statut>(`${this.apiUrl}/${codeStatut}`);
    }

    /**
     * Récupère tous les changements de statut d'une session
     */
    getChangementsParSession(sessionId: number): Observable<ChangementStatut[]> {
        return this.http.get<ChangementStatut[]>(`${this.changementsUrl}/session/${sessionId}`);
    }
}
